### PR TITLE
build: update node and yarn to latest stable versions

### PR DESCRIPTION
**Pull request by devversion**
Some CI jobs currently fail because db283f11eaf895e1e0b084a826cc9e3d6218e695 removed
Node v11 from the supported node versions. Some of CI jobs use Node v11 though, so they
currently fail. e.g. https://circleci.com/gh/angular/components/89692.

In order to fix this, we ensure that we use consistent versions. For Node, we use the current
active LTS version (v12). And for Yarn, we use the latest stable yarn version.

### CODE REVIEW

**Comment by devversion** [comment]
For reference: a few goldens had to be updated because the method overloads are now sorted based on their declaration order. This previously was not the case because `Array.sort` was not stable for more than 10 elements. see: https://bugs.chromium.org/p/v8/issues/detail?id=90

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,76 +7,74 @@
 # To validate changes, use an online parser, eg.
 # http://yaml-online-parser.appspot.com/
 
-var_1: &docker_image circleci/node:10.16
+var_1: &docker_image circleci/node:12.9.1
+var_2: &docker-firefox-image
+  # TODO(devversion): Temporarily use a image that includes Firefox 62 because the
+  # ngcontainer image does include an old Firefox version that does not support headless.
```
Comment (josephperrott) [inline .circleci/config.yml]: I think this reads a little better as `ngcontainer image includes an old Firefox`...

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
   },
   "license": "MIT",
   "engines": {
-    "node": "^10.13.0",
+    "node": "^12.0.0",
     "yarn": ">= 1.17.3"
```
Comment (josephperrott) [inline package.json]: Lets up update our requirement of yarn here as well.

File: .circleci/config.yml [hunk 3]
```diff
@@ -7,76 +7,74 @@
...
     command: yarn install --frozen-lockfile --non-interactive
 
 # Anchor that can be used to download and install Yarn globally in the bash environment.
-var_10: &yarn_download
+var_11: &yarn_download
   run:
     name: "Downloading and installing Yarn"
     command: |
       touch $BASH_ENV
       curl -o- -L https://yarnpkg.com/install.sh | PROFILE=$BASH_ENV bash -s -- --version "1.17.3"
```
Comment (josephperrott) [inline .circleci/config.yml]: We need to update our yarn version here as well

File: .circleci/config.yml (hunk 1 above)
Comment (devversion) [reply .circleci/config.yml]: I removed the whole comment since both images now match and it's no longer a workaround.

**Comment by devversion** [comment]
@josephperrott Feedback addressed. Please have another look

**Comment by Splaktar** [comment]
How is requiring NodeJS 12 working with relation to the fact that [Gulp 3 doesn't work](https://github.com/gulpjs/gulp/issues/2324) with it?

We're still on `"gulp": "^3.9.1",`.

**Pull request merged**